modules/runner/src/test/java/org/apache/ignite/internal/configuration/compatibility/framework/ConfigNode.java

```diff
@@ -41,29 +42,40 @@ public class ConfigNode {
     @JsonProperty
     private List<ConfigAnnotation> annotations = new ArrayList<>();
     @JsonProperty
-    private Map<String, ConfigNode> childNodeMap = new LinkedHashMap<>();
+    private final Map<String, ConfigNode> childNodeMap = new LinkedHashMap<>();
     @JsonProperty
     private String flagsHexString;
+    @JsonProperty
+    private Set<String> legacyPropertyNames;
 
     // Non-serializable fields.
     @JsonIgnore
-    private ConfigNode parent;
+    @Nullable private ConfigNode parent;
     @JsonIgnore
     private EnumSet<Flags> flags;
 
+    public static final ConfigNode INSTANCE = new ConfigNode();
```

Comment:
The purpose of this field is unclear.

Seems, it is used in single place as a poison object.
Let's move this to the place where it is used, rename to `DELETED` and write a javadoc....

Reply:
moved